src/repr/src/row.rs

```diff
@@ -375,8 +376,20 @@ unsafe fn read_datum<'a>(data: &'a [u8], offset: &mut usize) -> Datum<'a> {
         Tag::JsonNull => Datum::JsonNull,
         Tag::Dummy => Datum::Dummy,
         Tag::APD => {
-            let n = read_copy::<OrderedDecimal<Apd>>(data, offset);
-            Datum::APD(n)
+            let digits = read_copy::<u8>(data, offset);
+            let exponent = read_copy::<i8>(data, offset);
+            let bits = read_copy::<u8>(data, offset);
+            let lsu = read_untagged_bytes(data, offset);
+            // SAFETY: `lsu`'s data originated as a `&[u16]`, before being
+            // transmuted into a `&[u8]`; this transmute returns it to its
+            // original form.
+            let (prefix, lsu, suffix) = lsu.align_to::<u16>();
```

Comment:
This seems a little fishy to me. I think it’s possible that LSU gets written into a row in a way that it is not aligned to a two byte boundary. (E.g. if written to a row with a one byte string before the decimal.) It might actually be easiest if to_raw_parts and from_raw_parts dealt in u8 slices instead of u16 slices because that’s a very natural point for the copy into a properly aligned type to occur.

Reply:
believe I fixed this by virtue of MaterializeInc/rust-dec#42 + other minor changes